src/checks/stylelintConfigIsValid.ts

```diff
@@ -0,0 +1,102 @@
+import { promises as fs } from 'node:fs';
+import { createRequire } from 'node:module';
+import { execaCommand } from 'execa';
+import readdirp from 'readdirp';
+import semver from 'semver';
+import { projectPackageJson } from '../util/packageJson';
+import { supportedExtensions } from './stylelint';
+
+const require = createRequire(`${process.cwd()}/`);
+const projectDependencies = projectPackageJson.dependencies || {};
+
+export const title = 'Stylelint config is latest version';
+
+export default async function stylelintConfigIsValid() {
+  // continue only if project is Next.js or UpLeveled React App
+  if (
+    !(
+      '@upleveled/react-scripts' in projectDependencies ||
+      'next' in projectDependencies
+    )
+  ) {
+    return;
+  }
+
+  const { stdout: remoteVersion } = await execaCommand(
+    'npm show stylelint-config-upleveled version',
+  );
+
+  let localVersion;
+
+  try {
+    const stylelintConfigPackageJsonPath = require.resolve(
+      'stylelint-config-upleveled/package.json',
+    );
+
+    localVersion = JSON.parse(
+      await fs.readFile(stylelintConfigPackageJsonPath, 'utf-8'),
+    ).version;
+  } catch (err) {}
+
+  if (typeof localVersion === 'undefined') {
+    throw new Error(
+      `The UpLeveled Stylelint Config has not been installed. Please install using the instructions on https://www.npmjs.com/package/eslint-config-upleveled
+      `,
+    );
+  }
+
+  if (semver.gt(remoteVersion, localVersion)) {
+    throw new Error(
+      `Your current version of the UpLeveled Stylelint Config (${localVersion}) is older than the latest version ${remoteVersion} - upgrade by running:
+
+      pnpm install stylelint-config-upleveled@${remoteVersion}`,
+    );
+  }
+
+  let stylelintConfigMatches;
+
+  try {
+    stylelintConfigMatches =
+      (await fs.readFile('./stylelint.config.cjs', 'utf-8')).trim() ===
+      `/** @type { import('stylelint').Config } */
+const config = {
+  extends: ['stylelint-config-upleveled'],
+};
+
+module.exports = config;`;
+  } catch (err) {
+    throw new Error(
+      `Error reading your stylelint.config.cjs file - please delete the file if exists and reinstall the config using the instructions on https://www.npmjs.com/package/eslint-config-upleveled
+      `,
+    );
+  }
+
+  if (!stylelintConfigMatches) {
+    throw new Error(
+      `Your Stylelint config file stylelint.config.cjs does not match the configuration file template - please delete the file if exists and reinstall the config using the instructions on https://www.npmjs.com/package/eslint-config-upleveled
```

Comment:
this sentence
`Your Stylelint config file stylelint.config.cjs`
may be improved
maybe 
Your stylelint.config.cjs file does not ....

Reply:
1c086c7